### DISCUSSION TITLE
Add evaluation inside training script

### DISCUSSION THREAD

**Discussion by segalinc** [PixArt-alpha/PixArt-alpha#79] [Ideas]
I would be interesting to add an evaluation part in the train.py script to also long in the tracker.
This will generate images based on save_image_epoch config and log them in tensorboard or desired tracker  providing the prompts. Similar to the inference part but without the need to convert checkpoint to diffusers and load a pipeline

**Comment by DuoLi1999** [comment]
Have you implemented it yet? It would be fantastic if we could generate intermediate images during the training process.

**Comment by segalinc** [comment]
Yes I did. Might be helpful to double check I did it right.
Maybe I can create a PR for this or upload my script. I also added logging
to comet-ml if that's something that can be useful.
My script used precomputed latents tho so it may be a bit different from
the base code.

**Comment by lawrence-cj** [comment]
Check this one: https://github.com/PixArt-alpha/PixArt-alpha/blob/master/train_scripts/train_diffusers.py

**Comment by segalinc** [comment]
this is my version https://gist.github.com/segalinc/dca2ecd8ce3abb1c01ad231f36e28fd4

**Reply by lawrence-cj** [reply]
Cool. I like it. Have you tested it throughly? And would you mind pull a PR for this changes?

**Comment by segalinc** [comment]
Sent PR
